examples/knative/powershell/kn-ps-webhook/README.md

```diff
@@ -0,0 +1,197 @@
+# kn-ps-webhook
+Example Knative PowerShell function that accepts a custom incoming webhook, constructs a CloudEvent and sends it to the VMware Event Router (`broker`).
+
+![](screenshots/screenshot-01.png)
+
+The `CloudEvent` will be constructed with the following parameters and values:
+
+* **Source:** - Hostname (FQDN)
+* **Timestamp:** - Current Time of incoming request
+* **ID:** - v4 UUID
+* **Type:** - `com.vmware.generic.webhook.v0` (configurable via configuration secret/ENV variable)
+* **Subject:** (not-set)
+* **Data:** - Incoming body payload (body will be encoded using UTF8 and is expected to be JSON format)
+
+## Prerequisites
+
+* VEBA deployed with a [wildcard DNS entry](https://en.wikipedia.org/wiki/Wildcard_DNS_record), which is required for the ability to externally invoke a Knative Service URL that is serving the custom webhook function. Please [refer to this blog post](https://williamlam.com/2021/09/custom-webhook-function-to-publish-events-into-vmware-event-broker-appliance-veba.html) for more information.
+
+
+# Step 1 - Build
+
+Create the container image locally to test your function logic.
+
+```console
+export TAG=<version>
+docker build -t <docker-username>/kn-ps-webhook:${TAG} .
+```
+
+# Step 2 - Test
+
+Verify the container image works by executing it locally.
+
+Change into the `test` directory
+```console
+cd test
+```
+
+Update the following variable names within the `docker-test-env-variable` file and either the `send-cloudevent-test.ps1` (PowerShell Script) or `send-cloudevent-test.sh` (Bash Script)
+
+* WEBHOOK_USERNAME - Username to login to webhook function endpoint
+* WEBHOOK_PASSWORD - Password to login to webhook function endpoint
+* WEBHOOK_CE_EVENT_TYPE - The type value used for creating the CloudEvent (e.g. com.vmware.generic.webhook.v0)
+* WEBHOOK_SINK_URL - The sink URL (e.g. http://default-broker-ingress.vmware-functions.svc.cluster.local) to send the CloudEvent (leave blank for local development)
+
+Start the container image by running the following command:
+
+```console
+docker run -e SERVICE_DEBUG=true -e PORT=8080 --env-file docker-test-env-variable -it --rm -p 8080:8080 <docker-username>/kn-ps-webhook:${TAG}
+```
+
+In a separate terminal, run either `send-cloudevent-test.ps1` (PowerShell Script) or `send-cloudevent-test.sh` (Bash Script) to simulate a payload being sent to the local container image
```

Comment:
Should we rename the test files to something like send-webhook... because you don't send a CloudEvent